Merge 1.0.3 release

Areson opened the pull request:
Updating to prep for a new release to the forge.

Commit:
needed to replace the "::" values with "-" because of an issue with updated xml standards... at least in the eyes of the ReXML used in updated Ruby Interpreters in Puppet Agents.

Areson committed:
Merge branch 'master' into xml_purgeformatupdate

Areson committed:
Update CHANGELOG.md

Areson committed:
Update xml_fragment.rb

Reverting changes to hash item names in `self.xml_fragment_classvars`

alexjfisher commented on CHANGELOG.md:
```diff
@@ -1,8 +1,13 @@
+## 2019-??-?? - 1.0.3
+  - Added items to satisyf puppet-lin and pdk (Thanks alexjfisher)
```
Was this really me?

Areson replied:
Shoot, you are right. It was silug.

alexjfisher replied:
and I think I've found another bug.  Just debugging it now.

Areson replied:
Quite possible. There was a slew of pull requests and at least one broke a few things. We got a correction back in for that, but it's possible other scenarios that the reporter had not reviewed were failing.

alexjfisher replied:
```
undefined method `key?' for nil:NilClass
/opt/puppetlabs/puppet/cache/lib/puppet/util/xml_file.rb:136:in `block in set_tag'
/opt/puppetlabs/puppet/lib/ruby/2.5.0/rexml/xpath.rb:68:in `each'
/opt/puppetlabs/puppet/lib/ruby/2.5.0/rexml/xpath.rb:68:in `each'
/opt/puppetlabs/puppet/cache/lib/puppet/util/xml_file.rb:112:in `set_tag'
/opt/puppetlabs/puppet/cache/lib/puppet/provider/xml_fragment/xml_fragment.rb:154:in `create'
```

I don't think this is necessarily a regression in the pdk PR though. In fact, I'm pretty sure it's not.  I'll open a PR with a fix.

alexjfisher replied:
https://github.com/Areson/xml_fragment/pull/12

Areson replied:
That one isn't, no. One of the previous PRs modified some class variable names and it wasn't propagated throughout the code properly and broke some lookups.

alexjfisher replied:
yes, I saw that.  I've only hit these issues today as I'm trying to use `purge` for the first time.

alexjfisher commented on CHANGELOG.md:
```diff
@@ -1,8 +1,13 @@
+## 2019-??-?? - 1.0.3
+  - Added items to satisyf puppet-lin and pdk (Thanks alexjfisher)
+  - Added better validation for verifying proper parameters passed (Thanks alexjfisher)
+  - Changed syntax on attributes used by module to satisfy REXML (Thanks MacGregor001)
```
This changed isn't actually merged (or in this branch).

Areson replied:
No, not yet. I was just staging these to get ready to wrap this up to into a release to push to the forge. The changes are elsewhere and I was using this branch as a placeholder.

alexjfisher committed:
Fix `undefined method `key?' for nil:NilClass` in `set_tag`

alexjfisher committed:
Support puppet 6

Areson committed:
Merge pull request #13 from alexjfisher/puppet6

Support puppet 6

Areson committed:
Merge pull request #12 from alexjfisher/undefined_method

Fix `undefined method `key?' for nil:NilClass` in `set_tag`

Areson committed:
Merge pull request #6 from MacGregor001/xml_purgeformatupdate

needed to replace the "::" values with "-" because of an issue with u…

Areson committed:
Prevent File Save on every purge_tags call

Adjusted purge_tags method to prevent the file from being saved unless there were changes.

alexjfisher commented:
Any update on a release?

Areson commented:
I'll try and get it out in the next few days. I'm super behind on a number of projects because at some point all of my GitHub notifications got routed to my spam folder and I didn't notice.

Areson committed:
Update CHANGELOG.md

Finalizing changelog for release.

Areson committed:
Merge pull request #15 from Areson/development

Development

Areson commented:
Merged changes from the development branch into the release branch and prepping for final merge.
